pkg/controllers/tagging/tagging_controller.go

```diff
@@ -277,11 +282,19 @@ func (tc *Controller) process() bool {
 
 // tagNodesResources tag node resources
 // If we want to tag more resources, modify this function appropriately
-func (tc *Controller) tagNodesResources(node *v1.Node) error {
+func (tc *Controller) tagNodesResources(node *taggingControllerNode) error {
 	for _, resource := range tc.resources {
 		switch resource {
 		case opt.Instance:
-			err := tc.tagEc2Instance(node)
+			v1node, err := tc.nodeInformer.Lister().Get(node.name)
```

Comment:
Do we want to try to capture all necessary information in taggingControllerNode itself to avoid this lookup?

Reply:
(in a follow up!)

Reply:
we need to have a lookup if we want to avoid having multiple items in workqueue since the labels can be different on different updates.

Reply:
The `*Node` should point to the same struct though, right? the watcher/lister should only have one copy of the Node in memory